src/runtime_src/core/edge/drm/zocl/zocl_drv.c

```diff
@@ -371,12 +372,12 @@ static int zocl_mmap(struct file *filp, struct vm_area_struct *vma)
 		return -EINVAL;
 	}
 
-	phy_addr = vma->vm_pgoff << PAGE_SHIFT;
-
 	/* Only allow user to map register ranges in apertures list.
 	 * Could not map from the middle of an aperture.
 	 */
-	apt_idx = get_apt_index(zdev, phy_addr);
+	apt_idx = vma->vm_pgoff;
```

Comment:
Check if apt_idx is valid.

if (apt_idx >= zdev->num_apts)
        // Bad apt_idx, map failed.

Reply:
updated code